syncer/peers.go

```diff
@@ -0,0 +1,77 @@
+package syncer
+
+import (
+	"math/big"
+	"sync"
+
+	"github.com/libp2p/go-libp2p-core/peer"
+)
+
+type NoForkPeer struct {
+	// identifier
+	ID peer.ID
+	// peer's latest block number
+	Number uint64
+	// peer's distance
+	Distance *big.Int
+}
+
+func (p *NoForkPeer) IsBetter(t *NoForkPeer) bool {
+	if p.Number != t.Number {
+		return p.Number > t.Number
+	}
+
+	return p.Distance.Cmp(t.Distance) < 0
+}
+
+type PeerMap struct {
+	sync.Map
+}
+
+func NewPeerMap(peers []*NoForkPeer) *PeerMap {
+	peerMap := new(PeerMap)
+
+	peerMap.PutPeers(peers)
+
+	return peerMap
+}
+
+func (m *PeerMap) PutPeers(peers []*NoForkPeer) {
+	for _, p := range peers {
+		m.Put(p)
+	}
+}
+
+func (m *PeerMap) Put(peer *NoForkPeer) {
+	m.Store(peer.ID.String(), peer)
+}
+
+// Remove removes a peer from heap if it exists
+func (m *PeerMap) Remove(peerID peer.ID) {
+	m.Delete(peerID.String())
+}
+
+// BestPeer returns the top of heap
+func (m *PeerMap) BestPeer(skipMap map[peer.ID]bool) *NoForkPeer {
+	var bestPeer *NoForkPeer
+
+	m.Range(func(key, value interface{}) bool {
+		peer, ok := value.(*NoForkPeer)
+		if !ok {
+			// shouldn't reach here
+			return false
+		}
```

Comment:
I think you can just assume that the cast is ok

Reply:
I know. But I don't like adding `//nolint:forcetypeassert` 😂 That's what I don't like in Golang

Reply:
Fixed at d75dc06974a15bbf55ba2bc69f8173b95e531db2

Reply:
To trick the linter you can do:

`obj, _ := myCast.(castType)`

The `_` will fool it

Reply:
Fixed de1f1cd96b55bee48e6a1cd812f97f0e9e3108ae

Reply:
ping @zivkovicmilos

Reply:
This commit hash does not point to the fix

Reply:
https://github.com/0xPolygon/polygon-edge/pull/591/commits/03dc08ad39a048560211355b3335b53e63589de6